### PR TITLE
Add #9266 [v96] Shortcut undo last tab closed

### PR DESCRIPTION
**Pull request by lmarceau**
## Issue #9266
Add a keyboard shortcut to undo a tab that was closed.
Using the ClosedTabsStore which currently holds up to 10 closed tabs.
Added unit tests for ClosedTabsStore and fix deprecation warnings for NSKeyedArchiver and NSKeyedUnarchiver

### CODE REVIEW

File: Client/Frontend/Browser/BrowserViewController/BrowserViewController+KeyCommands.swift [hunk 1]
```diff
@@ -116,6 +116,15 @@ extension BrowserViewController {
         tabManager.removeTab(currentTab)
     }
 
+    @objc private func undoLastTabClosedKeyCommand() {
+        guard let lastClosedURL = profile.recentlyClosedTabs.popFirstTab()?.url,
+            let selectedTab = tabManager.selectedTab else { return }
```
Comment (electricRGB) [inline Client/Frontend/Browser/BrowserViewController/BrowserViewController+KeyCommands.swift:121]:
```suggestion
        guard let lastClosedURL = profile.recentlyClosedTabs.popFirstTab()?.url,
                   let selectedTab = tabManager.selectedTab
        else { return }
```

File: Storage/RecentlyClosedTabs.swift [hunk 2]
```diff
@@ -57,9 +75,9 @@ open class ClosedTab: NSObject, NSCoding {
     }
 
     required convenience public init?(coder: NSCoder) {
-        guard let url = coder.decodeObject(forKey: "url") as? URL,
-              let faviconURL = coder.decodeObject(forKey: "faviconURL") as? String,
-              let title = coder.decodeObject(forKey: "title") as? String else { return nil }
+        guard let url = coder.decodeObject(forKey: CodingKeys.url.rawValue) as? URL,
+              let faviconURL = coder.decodeObject(forKey: CodingKeys.faviconURL.rawValue) as? String,
+              let title = coder.decodeObject(forKey: CodingKeys.title.rawValue) as? String else { return nil }
```
Comment (electricRGB) [inline Storage/RecentlyClosedTabs.swift:80]:
```suggestion
              let title = coder.decodeObject(forKey: CodingKeys.title.rawValue) as? String
        else { return nil }
```

File: Storage/RecentlyClosedTabs.swift [hunk 3]
```diff
@@ -6,11 +6,17 @@ import Foundation
...
+    }
 
     lazy open var tabs: [ClosedTab] = {
-        guard let tabsArray: Data = self.prefs.objectForKey("recentlyClosedTabs") as Any? as? Data,
-              let unarchivedArray = NSKeyedUnarchiver.unarchiveObject(with: tabsArray) as? [ClosedTab] else {
+        guard let tabsArray: Data = self.prefs.objectForKey(KeyedArchiverKeys.recentlyClosedTabs.rawValue) as Any? as? Data,
+              let unarchiver = try? NSKeyedUnarchiver(forReadingFrom: tabsArray),
+              let unarchivedArray = unarchiver.decodeObject(of: [NSArray.self, ClosedTab.self], forKey: KeyedArchiverKeys.recentlyClosedTabs.rawValue) as? [ClosedTab] else {
             return []
         }
```
Comment (electricRGB) [inline Storage/RecentlyClosedTabs.swift:21]:
```suggestion
              let unarchivedArray = unarchiver.decodeObject(of: [NSArray.self, ClosedTab.self], forKey: KeyedArchiverKeys.recentlyClosedTabs.rawValue) as? [ClosedTab]
        else { return [] }
```

**Review by electricRGB — APPROVED** [review]

**Comment by lmarceau** [comment]
testURLBar test passes locally 
Screen Shot 2021-12-22 at 4 24 00 PM

**Pull request merged**